### DISCUSSION TITLE
Is there any documentation on error handling?

### DISCUSSION THREAD

**Discussion by ghost** [jgaeddert/liquid-dsp#325] [Q&A] [answered]
I am searching for documentation around error handling in liquid-dsp. I noticed that some create functions will exit the program on failure and many functions return status integers, but I can't find anything obvious in the documentation for interpreting or handling these.

**Comment by jgaeddert** [comment, marked as answer]
Great question! All objects' `create` methods will (eventually) return a NULL pointer upon error and log the particular error message. I realize that pointing to the massive header file is really not a valid answer, but the error codes are described here:

https://github.com/jgaeddert/liquid-dsp/blob/master/include/liquid.h#L76-L126

The intent here is to provide some basic level of indication for what went wrong and support debugging, but not really emulate exception handling in C. I'd love to hear any suggestions you might have to make this better!

**Reply by ghost** [reply]
Pointing to the header file is absolutely a valid answer! It turns out that liquid-dsp provided through apt is still at version 1.3.2 which doesn't seem to have these codes. I didn't realize it was so far behind.

I do have a suggestion, but I am just some guy so feel free to ignore 😅

SDL return error codes that can be used to lookup error messages, but they also set a global error string that can be accessed via SDL_GetError(). This allows the user to access the error message following a failed create-function which returns a null pointer instead of an error code. This is great because it allows the user to control when and how the error message is logged, I personally create an exception type that wraps the error message.

Anyway, thank you so much for the response, your work is amazing!
